tests/test_read.py

```diff
@@ -353,3 +353,86 @@ def test_mdvrptw_instance():
     assert_(any(depot.y < 0) for depot in data.depots())
     assert_(any(client.x < 0) for client in data.clients())
     assert_(any(client.y < 0) for client in data.clients())
+
+
+def test_vrpspd_instance():
+    """
+    Tests that reading an VRPSPD instance happens correctly, particularly the
+    backhaul data.
+    """
+    data = read("data/RC101.vrp")
+
+    assert_equal(data.num_locations, 5)
+    assert_equal(data.num_depots, 1)
+    assert_equal(data.num_clients, 4)
+
+    assert_equal(data.num_vehicles, 4)
+    assert_equal(data.num_vehicle_types, 1)
+
+    vehicle_type = data.vehicle_type(0)
+    assert_equal(vehicle_type.num_available, 4)
+    assert_equal(vehicle_type.capacity, 200)
+
+    # The first client is a linehaul client (only delivery, no pickup), and
+    # the second client is a backhaul client (only pickup, no delivery). All
+    # other clients have both delivery and pickup.
+    clients = data.clients()
+
+    assert_(clients[0].delivery > 0 and clients[0].pickup == 0)
+    assert_(clients[1].delivery == 0 and clients[1].pickup > 0)
+
+    for client in data.clients()[2:]:
+        assert_(client.pickup > 0 and client.delivery > 0)
+
+    # Test that distance/duration are not set to a large value, as in VRPB.
+    assert_(np.all(data.distance_matrix() <= 142))  # max distance in instance
+    assert_(np.all(data.duration_matrix() <= 142))  # max duration in instance
+
+
+def test_vrpb_instance():
+    """
+    Tests that reading an VRPB instance happens correctly, particularly the
+    backhaul data and modified distances to ensure linehaul is served before
+    backhaul.
+    """
+    data = read("data/X-n101-50-k13.vrp", round_func="round")
+
+    assert_equal(data.num_locations, 101)
+    assert_equal(data.num_depots, 1)
+    assert_equal(data.num_clients, 100)
+
+    assert_equal(data.num_vehicles, 100)
+    assert_equal(data.num_vehicle_types, 1)
+
+    vehicle_type = data.vehicle_type(0)
+    assert_equal(vehicle_type.num_available, 100)
+    assert_equal(vehicle_type.capacity, 206)
+
+    # The first 50 clients are linehaul, the rest are backhaul.
+    clients = data.clients()
+
+    for client in clients[:50]:
+        assert_(client.pickup == 0)
+        assert_(client.delivery > 0)
+
+    for client in clients[50:]:
+        assert_(client.pickup > 0)
+        assert_(client.delivery == 0)
+
+    # Tests that distance/duration from depot to backhaul clients is set to
+    # a large value, as well as for backhaul to linehaul clients.
+    linehauls = set(range(1, 51))
+    backhauls = set(range(51, 101))
+    max_value = 1_000_000
```

Comment:
This is supposed to be exactly `MAX_USER_VALUE`, right? We can import that from `pyvrp.constants` right? E.g. `from pyvrp.constants import MAX_USER_VALUE`, and then make the checks below exact using `assert_allclose` with `MAX_USER_VALUE` (where appropriate). All checks that should be smaller can remain inequalities, but this should provide some guarantees that we are indeed setting the correct value in the correct places.